Bug Report - 1.21 crashing on launch

**Describe the bug**
Mod causes the game to crash on launch

**Crash Logs**---- Minecraft Crash Report ----
// Hey, that tickles! Hehehe!

Time: 2024-07-26 15:38:27
Description: Initializing game

java.lang.RuntimeException: Could not execute entrypoint stage 'main' due to errors, provided by 'regions_unexplored'!
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:477)
	at net.minecraft.client.main.Main.main(Main.java:239)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)
	Suppressed: java.lang.NoClassDefFoundError: net/neoforged/fml/config/IConfigSpec
		at java.base/java.lang.Class.forName0(Native Method)
		at java.base/java.lang.Class.forName(Class.java:534)
		at java.base/java.lang.Class.forName(Class.java:513)
		at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
		at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
		at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
		at terrablender.core.TerraBlenderFabric.lambda$onInitialize$0(TerraBlenderFabric.java:35)
		at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
		at terrablender.core.TerraBlenderFabric.onInitialize(TerraBlenderFabric.java:34)
		at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
		... 6 more
	Caused by: java.lang.ClassNotFoundException: net.neoforged.fml.config.IConfigSpec
		at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
		at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
		at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
		at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
		... 16 more
Caused by: java.lang.NoClassDefFoundError: net/neoforged/fml/config/IConfigSpec
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:534)
	at java.base/java.lang.Class.forName(Class.java:513)
	at net.fabricmc.loader.impl.util.DefaultLanguageAdapter.create(DefaultLanguageAdapter.java:50)
	at net.fabricmc.loader.impl.entrypoint.EntrypointStorage$NewEntry.getOrCreate(EntrypointStorage.java:117)
	at net.fabricmc.loader.impl.entrypoint.EntrypointContainerImpl.getEntrypoint(EntrypointContainerImpl.java:53)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:384)
	... 6 more
Caused by: java.lang.ClassNotFoundException: net.neoforged.fml.config.IConfigSpec
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:641)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	at net.fabricmc.loader.impl.launch.knot.KnotClassDelegate.loadClass(KnotClassDelegate.java:226)
	at net.fabricmc.loader.impl.launch.knot.KnotClassLoader.loadClass(KnotClassLoader.java:119)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526)
	... 13 more


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.fabricmc.loader.impl.FabricLoaderImpl.lambda$invokeEntrypoints$2(FabricLoaderImpl.java:388)
	at net.fabricmc.loader.impl.util.ExceptionUtil.gatherExceptions(ExceptionUtil.java:33)
	at net.fabricmc.loader.impl.FabricLoaderImpl.invokeEntrypoints(FabricLoaderImpl.java:386)
	at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:52)
	at net.minecraft.class_310.<init>(class_310.java:477)

-- Initialization --
Details:
	Modules: 
		ADVAPI32.dll:Advanced Windows 32 Base API:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		COMCTL32.dll:User Experience Controls Library:6.10 (WinBuild.160101.0800):Microsoft Corporation
		CRYPT32.dll:Crypto API32:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTBASE.dll:Base cryptographic API DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		CRYPTSP.dll:Cryptographic Service Provider API:10.0.22621.3672 (WinBuild.160101.0800):Microsoft Corporation
		DBGHELP.DLL:Windows Image Helper:10.0.22621.3593 (WinBuild.160101.0800):Microsoft Corporation
		DNSAPI.dll:DNS Client API DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		GDI32.dll:GDI Client DLL:10.0.22621.3672 (WinBuild.160101.0800):Microsoft Corporation
		IMM32.DLL:Multi-User Windows IMM32 API Client DLL:10.0.22621.3374 (WinBuild.160101.0800):Microsoft Corporation
		IPHLPAPI.DLL:IP Helper API:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		KERNEL32.DLL:Windows NT BASE API Client DLL:10.0.22621.3958 (WinBuild.160101.0800):Microsoft Corporation
		KERNELBASE.dll:Windows NT BASE API Client DLL:10.0.22621.3958 (WinBuild.160101.0800):Microsoft Corporation
		NSI.dll:NSI User-mode interface DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		NTASN1.dll:Microsoft ASN.1 API:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		OLEAUT32.dll:OLEAUT32.DLL:10.0.22621.3672 (WinBuild.160101.0800):Microsoft Corporation
		Ole32.dll:Microsoft OLE for Windows:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		POWRPROF.dll:Power Profile Helper DLL:10.0.22621.3374 (WinBuild.160101.0800):Microsoft Corporation
		PSAPI.DLL:Process Status Helper:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		Pdh.dll:Windows Performance Data Helper DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		RPCRT4.dll:Remote Procedure Call Runtime:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		SHCORE.dll:SHCORE:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		SHELL32.dll:Windows Shell Common Dll:10.0.22621.3958 (WinBuild.160101.0800):Microsoft Corporation
		UMPDC.dll:User Mode Power Dependency Coordinator:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		USER32.dll:Multi-User Windows USER API Client DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		USERENV.dll:Userenv:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		VCRUNTIME140.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		VERSION.dll:Version Checking and File Installation Libraries:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WINHTTP.dll:Windows HTTP Services:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WINMM.dll:MCI API DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		WS2_32.dll:Windows Socket 2.0 32-Bit DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		amsi.dll:Anti-Malware Scan Interface:10.0.22621.3527 (WinBuild.160101.0800):Microsoft Corporation
		bcrypt.dll:Windows Cryptographic Primitives Library:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		bcryptPrimitives.dll:Windows Cryptographic Primitives Library:10.0.22621.3672 (WinBuild.160101.0800):Microsoft Corporation
		clbcatq.dll:COM+ Configuration Catalog:2001.12.10941.16384 (WinBuild.160101.0800):Microsoft Corporation
		combase.dll:Microsoft COM for Windows:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		dbgcore.DLL:Windows Core Debugging Helpers:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		extnet.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		fwpuclnt.dll:FWP/IPsec User-Mode API:10.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		gdi32full.dll:GDI Client DLL:10.0.22621.3958 (WinBuild.160101.0800):Microsoft Corporation
		java.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		javaw.exe:OpenJDK Platform binary:21.0.3.0:Microsoft
		jemalloc.dll
		jimage.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		jli.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		jna1478344540689986043.dll:JNA native library:7.0.0:Java(TM) Native Access (JNA)
		jsvml.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		jvm.dll:OpenJDK 64-Bit server VM:21.0.3.0:Microsoft
		kernel.appcore.dll:AppModel API Host:10.0.22621.3958 (WinBuild.160101.0800):Microsoft Corporation
		lwjgl.dll
		management.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		management_ext.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		msvcp140.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		msvcp_win.dll:Microsoft® C Runtime Library:10.0.22621.3374 (WinBuild.160101.0800):Microsoft Corporation
		msvcrt.dll:Windows NT CRT DLL:7.0.22621.2506 (WinBuild.160101.0800):Microsoft Corporation
		mswsock.dll:Microsoft Windows Sockets 2.0 Service Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		napinsp.dll:E-mail Naming Shim Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		ncrypt.dll:Windows NCrypt Router:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		net.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		nio.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		nlansp_c.dll:NLA Namespace Service Provider DLL:10.0.22621.3672 (WinBuild.160101.0800):Microsoft Corporation
		ntdll.dll:NT Layer DLL:10.0.22621.3374 (WinBuild.160101.0800):Microsoft Corporation
		perfos.dll:Windows System Performance Objects DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		pfclient.dll:SysMain Client:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		pnrpnsp.dll:PNRP Name Space Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		profapi.dll:User Profile Basic API:10.0.22621.3527 (WinBuild.160101.0800):Microsoft Corporation
		rasadhlp.dll:Remote Access AutoDial Helper:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		rsaenh.dll:Microsoft Enhanced Cryptographic Provider:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		sechost.dll:Host for SCM/SDDL/LSA Lookup APIs:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		shlwapi.dll:Shell Light-weight Utility Library:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		sunmscapi.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		ucrtbase.dll:Microsoft® C Runtime Library:10.0.22621.3593 (WinBuild.160101.0800):Microsoft Corporation
		vcruntime140_1.dll:Microsoft® C Runtime Library:14.29.30139.0 built by: vcwrkspc:Microsoft Corporation
		verify.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
		win32u.dll:Win32u:10.0.22621.3958 (WinBuild.160101.0800):Microsoft Corporation
		windows.storage.dll:Microsoft WinRT Storage API:10.0.22621.3527 (WinBuild.160101.0800):Microsoft Corporation
		winrnr.dll:LDAP RnR Provider DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		wintypes.dll:Windows Base Types DLL:10.0.22621.1 (WinBuild.160101.0800):Microsoft Corporation
		wshbth.dll:Windows Sockets Helper DLL:10.0.22621.3958 (WinBuild.160101.0800):Microsoft Corporation
		zip.dll:OpenJDK Platform binary:21.0.3.0:Microsoft
Stacktrace:
	at net.minecraft.client.main.Main.main(Main.java:239)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:470)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- System Details --
Details:
	Minecraft Version: 1.21
	Minecraft Version ID: 1.21
	Operating System: Windows 11 (amd64) version 10.0
	Java Version: 21.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 327565904 bytes (312 MiB) / 570425344 bytes (544 MiB) up to 6442450944 bytes (6144 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
	Identifier: Intel64 Family 6 Model 165 Stepping 2
	Microarchitecture: Comet Lake
	Frequency (GHz): 2.59
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: Intel(R) UHD Graphics
	Graphics card #0 vendor: Intel Corporation
	Graphics card #0 VRAM (MiB): 1024.00
	Graphics card #0 deviceId: VideoController1
	Graphics card #0 versionInfo: 30.0.101.1994
	Graphics card #1 name: NVIDIA GeForce RTX 2060 with Max-Q Design
	Graphics card #1 vendor: NVIDIA
	Graphics card #1 VRAM (MiB): 6144.00
	Graphics card #1 deviceId: VideoController2
	Graphics card #1 versionInfo: 31.0.15.4626
	Memory slot #0 capacity (MiB): 16384.00
	Memory slot #0 clockSpeed (GHz): 3.20
	Memory slot #0 type: DDR4
	Virtual memory max (MiB): 33568.11
	Virtual memory used (MiB): 21553.19
	Swap memory total (MiB): 17408.00
	Swap memory used (MiB): 273.59
	Space in storage for jna.tmpdir (MiB): available: 56275.81, total: 242829.00
	Space in storage for org.lwjgl.system.SharedLibraryExtractPath (MiB): available: 56275.81, total: 242829.00
	Space in storage for io.netty.native.workdir (MiB): available: 56275.81, total: 242829.00
	Space in storage for java.io.tmpdir (MiB): available: 56275.81, total: 242829.00
	Space in storage for workdir (MiB): available: 579525.25, total: 953867.00
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx6G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		fabric-api: Fabric API 0.100.7+1.21
			fabric-api-base: Fabric API Base 0.4.42+6573ed8cd1
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.67+b5597344d1
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.29+5bd9f1bcd1
			fabric-block-api-v1: Fabric Block API (v1) 1.0.22+0af3f5a7d1
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.10+6573ed8cd1
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.52+0af3f5a7d1
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.15+6573ed8cd1
			fabric-command-api-v1: Fabric Command API (v1) 1.2.49+f71b366fd1
			fabric-command-api-v2: Fabric Command API (v2) 2.2.28+6ced4dd9d1
			fabric-commands-v0: Fabric Commands (v0) 0.2.66+df3654b3d1
			fabric-content-registries-v0: Fabric Content Registries (v0) 8.0.15+b5597344d1
			fabric-convention-tags-v1: Fabric Convention Tags 2.0.18+7f945d5bd1
			fabric-convention-tags-v2: Fabric Convention Tags (v2) 2.4.2+c111832ad1
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.29+0af3f5a7d1
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.1.25+6a6dfa19d1
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 20.2.12+16c4ae25d1
			fabric-dimensions-v1: Fabric Dimensions API (v1) 4.0.0+6fc22b99d1
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.6.12+6fc22b99d1
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.7.10+e633f883d1
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.53+6ced4dd9d1
			fabric-item-api-v1: Fabric Item API (v1) 11.0.0+afdfc921d1
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.1.3+78017270d1
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.47+0af3f5a7d1
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.45+df3654b3d1
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.3.11+8f3583aed1
			fabric-loot-api-v2: Fabric Loot API (v2) 3.0.13+3f89f5a5d1
			fabric-loot-api-v3: Fabric Loot API (v3) 1.0.1+3f89f5a5d1
			fabric-message-api-v1: Fabric Message API (v1) 6.0.13+6573ed8cd1
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 2.0.0+fe474d6bd1
			fabric-networking-api-v1: Fabric Networking API (v1) 4.2.0+ab7edbacd1
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 15.1.13+d1321076d1
			fabric-particles-v1: Fabric Particles (v1) 4.0.2+6573ed8cd1
			fabric-recipe-api-v1: Fabric Recipe API (v1) 5.0.10+65089712d1
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 5.0.25+5bd9f1bcd1
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.3.0+0ae0b97dd1
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.6.5+48fb1586d1
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.68+df3654b3d1
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.48+73761d2ed1
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.1.6+b5597344d1
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.71+df3654b3d1
			fabric-rendering-v1: Fabric Rendering (v1) 5.0.5+df16efd0d1
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 4.2.1+d153f344d1
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 1.2.0+7e687b32d1
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.24+b5597344d1
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.82+b5597344d1
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.23+6573ed8cd1
			fabric-transfer-api-v1: Fabric Transfer API (v1) 5.1.15+3dccd343d1
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 6.0.12+6573ed8cd1
		fabricloader: Fabric Loader 0.15.11
			mixinextras: MixinExtras 0.3.5
		java: OpenJDK 64-Bit Server VM 21
		minecraft: Minecraft 1.21
		regions_unexplored: Regions Unexplored 0.5.6
		terrablender: TerraBlender 4.0.0.0
			com_electronwill_night-config_core: core 3.6.7
			com_electronwill_night-config_toml: toml 3.6.7
	Launched Version: fabric-loader-0.15.11-1.21
	Launcher name: minecraft-launcher
	Backend library: LWJGL version 3.3.3-snapshot
	Backend API: Unknown
	Window size: <not initialized>
	GFLW Platform: <error>
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: <disabled>
	Is Modded: Definitely; Client brand changed to 'fabric'
	Universe: 404
	Type: Client (map_client.txt)
	Locale: en_US
	System encoding: Cp1252
	File encoding: UTF-8
	CPU: <unknown>
**How To Reproduce**
launched game with only terrablender and fabric api mods

**Version**
Minecraft Version: 1.21

Modloader Version: fabric-loader-0.15.11-1.21

Mod Version: v0.5.6

**Screenshots**
![image](https://github.com/user-attachments/assets/a9c125a5-c7bf-40d1-80c1-b5e788e5af74)

--- a/common/src/main/java/net/regions_unexplored/client/particle/RuParticleTypes.java
+++ b/common/src/main/java/net/regions_unexplored/client/particle/RuParticleTypes.java
@@ -6,15 +6,15 @@ import net.minecraft.core.registries.BuiltInRegistries;
 import net.regions_unexplored.platform.Services;
 
 public class RuParticleTypes {
-    public static final SimpleParticleType MAUVE_LEAVES = new SimpleParticleType(false);
-    public static final SimpleParticleType MYCOTOXIC_SPORE = new SimpleParticleType(false);
-    public static final SimpleParticleType BLUE_MAGNOLIA_LEAVES = new SimpleParticleType(false);
-    public static final SimpleParticleType ENCHANTED_BIRCH_LEAVES = new SimpleParticleType(false);
-    public static final SimpleParticleType ORANGE_MAPLE_LEAVES = new SimpleParticleType(false);
-    public static final SimpleParticleType PINK_MAGNOLIA_LEAVES = new SimpleParticleType(false);
-    public static final SimpleParticleType RED_MAPLE_LEAVES = new SimpleParticleType(false);
-    public static final SimpleParticleType SILVER_BIRCH_LEAVES = new SimpleParticleType(false);
-    public static final SimpleParticleType WHITE_MAGNOLIA_LEAVES = new SimpleParticleType(false);
+    public static final SimpleParticleType MAUVE_LEAVES = simple(true);
+    public static final SimpleParticleType MYCOTOXIC_SPORE = simple(true);
+    public static final SimpleParticleType BLUE_MAGNOLIA_LEAVES = simple(true);
+    public static final SimpleParticleType ENCHANTED_BIRCH_LEAVES = simple(true);
+    public static final SimpleParticleType ORANGE_MAPLE_LEAVES = simple(true);
+    public static final SimpleParticleType PINK_MAGNOLIA_LEAVES = simple(true);
+    public static final SimpleParticleType RED_MAPLE_LEAVES = simple(true);
+    public static final SimpleParticleType SILVER_BIRCH_LEAVES = simple(true);
+    public static final SimpleParticleType WHITE_MAGNOLIA_LEAVES = simple(true);
 
     public static void addParticles() {
         register("mauve_leaves", MAUVE_LEAVES);
@@ -30,5 +30,9 @@ public class RuParticleTypes {
 
     private static void register(String name, ParticleType<?> particleType) {
         Services.REGISTAR.register(BuiltInRegistries.PARTICLE_TYPE, name, () -> particleType);
+    }
+
+    public static SimpleParticleType simple(boolean alwaysSpawn) {
+        return new SimpleParticleType(alwaysSpawn) { };
     }
 }

--- a/fabric/src/main/resources/fabric.mod.json
+++ b/fabric/src/main/resources/fabric.mod.json
@@ -32,11 +32,13 @@
         "${mod_id}.mixins.json",
         "${mod_id}.fabric.mixins.json"
     ],
+    "accessWidener": "regions_unexplored.accesswidener",
     "depends": {
         "fabricloader": ">=${fabric_loader_version}",
         "fabric-api": "*",
         "minecraft": "${minecraft_version}",
-        "java": ">=${java_version}"
+        "java": ">=${java_version}",
+        "forgeconfigapiport": ">=21"
     }
 }
   

